Zend/zend_vm_def.h

```diff
@@ -8039,6 +8039,39 @@ ZEND_VM_HANDLER(51, ZEND_MAKE_REF, VAR|CV, UNUSED)
 	ZEND_VM_NEXT_OPCODE();
 }
 
+ZEND_VM_HANDLER(187, ZEND_SWITCH, CONST|TMPVAR|CV, CONST, JMP_ADDR)
+{
+	USE_OPLINE
+	zend_free_op free_op1, free_op2;
+	zval *op, *table_op, *jump_zv;
+	HashTable *ht;
+
+	op = GET_OP1_ZVAL_PTR_UNDEF(BP_VAR_R);
+	table_op = GET_OP2_ZVAL_PTR(BP_VAR_R);
+	ht = Z_ARRVAL_P(table_op);
+
+	if (Z_TYPE_P(op) != opline->result.num) {
+		/* Wrong type, fall back to ZEND_CASE chain */
```

Comment:
All the case types are uniform, but we still have to check that the value passed to `switch` has the same type. E.g. for something like `switch (1.5) { case 1: ... }` you have `opline->result.num == IS_LONG` and `Z_TYPE_P(op) == IS_DOUBLE`. In this case it falls back to normal implementation using `CASE` checks.